File: src/main/java/sk/hudak/prco/utils/ConvertUtils.java
package sk.hudak.prco.utils;

import sk.hudak.prco.exception.StringToNumberConvertPrcoException;

import java.math.BigDecimal;

import static sk.hudak.prco.utils.Validate.notNullNotEmpty;

public class ConvertUtils {

    public static final String COMMA = ",";
    public static final String DOT = ".";

    private ConvertUtils() {
    }

    public static BigDecimal convertToBigDecimal(String value) {
        notNullNotEmpty(value, "value");
        try {
            return new BigDecimal(value.replace(COMMA, DOT));

        } catch (Exception e) {
            throw new StringToNumberConvertPrcoException(value, e);
        }
    }
}

File: src/main/java/sk/hudak/prco/utils/ConvertUtils.kt
package sk.hudak.prco.utils

import sk.hudak.prco.exception.StringToNumberConvertPrcoException
import sk.hudak.prco.utils.Validate.notNullNotEmpty
import java.math.BigDecimal

object ConvertUtils {

    private const val COMMA = ","
    private const val DOT = "."

    //TODO uribit ako extenzions  pre String class !!!
    @JvmStatic
    fun convertToBigDecimal(value: String): BigDecimal {
        notNullNotEmpty(value, "value")

        try {
            return BigDecimal(value.replace(COMMA, DOT))

        } catch (e: Exception) {
            throw StringToNumberConvertPrcoException(value, e)
        }

    }
}
